### ISSUE TITLE
Allow CustomValueComparators to handle nulls

### ISSUE THREAD

**Issue by bartoszwalacik** [javers/javers#1075]
Currently, before calling a CustomValueComparator impl, Javers checks nulls and assumes that :

```
null != any non-null value
```

This assumption simplifies an impl, but from the other hand it 
makes impossible to write a Comparator which treats nulls as empty values.
See https://stackoverflow.com/questions/66497652/javers-comparison-string-collection-and-boolean-with-defined-rules

```java
public interface CustomValueComparator<T> {
    /**
     * Called by JaVers to compare two Values.
     *
     * @param a not null
     * @param b not null
     */
    boolean equals(T a, T b);
```

**New feature**
Skip the Javers internal null-safety logic and allow passing nulls to CustomValueComparators. 

Unfortunately it could break existing users' code

**Test case**
[CaseWithCustomComparatorBlejwas.groovy](https://github.com/javers/javers/commit/045d751f4b7a8cdb959b246bc9b85538993befc2)

**Comment by djavaphp** [comment]
Hi 
do you have any update on this? 
I want to ignore upcoming fields which is empty or null. 
Is there any way to do?

**Comment by bartoszwalacik** [comment]
Hi, if you want to speed up things, feel free to contribute a PR

**Comment by ratskinmahoney** [comment]
@bartoszwalacik - I'd be happy to raise a PR for this, but wanted to check if you had any thoughts on approach before I do.

Currently, what I am thinking is that I would modify `CustomValueComparator` to include a method to indicate whether wrapping with the `CustomValueComparatorNullSafe` should be skipped (with a default that returns false):

```java
public interface CustomValueComparator<T> {
  
  // ...

  default boolean skipNullSafety() { return false; }
}
```

This could then be extended with an interface that clearly describes it's behaviour:

```java
public interface NonNullSafeCustomValueComparator<T> extends CustomValueComparator<T> {

  @Override
  boolean skipNullSafety() { return true; }
}
```

Then `PrimitiveOrValueType` can be modified to check this flag:

```java
  PrimitiveOrValueType(Type baseJavaType, CustomValueComparator<T> comparator) {
    super(baseJavaType);
    this.valueComparator = comparator == null || comparator.skipNullSafety() ? comparator : new CustomValueComparatorNullSafe<>(comparator);
  }
```

That seems simple enough, as well as backwards compatible. I'm not sure if the new comparator interface should be advertised more clearly (e.g. through a new method on the builder, but that seems a bit bloating).

Any thoughts or concerns?

Thanks very much for your work, by the way. Javers has saved me many, many hours of effort over recent months!

**Comment by bartoszwalacik** [comment]
Ok @ratskinmahoney, your design looks good, but maybe we can avoid double negations?

```
public interface CustomValueComparator<T> {
  
  // ...

  default boolean nullSafety() { return true; }
}
```

**Comment by ratskinmahoney** [comment]
Thanks @bartoszwalacik - and good point. I'll make the naming more straightforward as suggested. I'll get a PR raised presently.

**Comment by ratskinmahoney** [comment]
I've raised PR #1159. I've adjusted some of the naming so we refer to null handling, rather than null safety as it seemed to make things a bit clearer. Perfectly happy if we'd prefer to use the naming suggested above.

**Comment by bartoszwalacik** [comment]
fixed in 6.5.2
